src/dxtbx/model/goniometer.py

```diff
@@ -392,69 +392,69 @@ def imgCIF_H(cbf_handle):
         it is assumed that the file has already been read."""
 
         # find the goniometer axes and dependencies
-        axis_names = flex.std_string()
-        depends_on = flex.std_string()
-        axes = flex.vec3_double()
-        angles = flex.double()
+        dependants = {}
+        axis_vectors = {}
+        angles = {}
         scan_axis = None
         cbf_handle.find_category(b"axis")
         for i in range(cbf_handle.count_rows()):
             cbf_handle.find_column(b"equipment")
             if cbf_handle.get_value() == b"goniometer":
                 cbf_handle.find_column(b"id")
-                axis_names.append(cbf_handle.get_value())
-                axis = []
+                axis_name = cbf_handle.get_value().decode()
+                axis_vector = []
                 for i in range(3):
                     cbf_handle.find_column(b"vector[%i]" % (i + 1))
-                    axis.append(float(cbf_handle.get_value()))
-                axes.append(axis)
+                    axis_vector.append(float(cbf_handle.get_value()))
+                axis_vectors[axis_name] = axis_vector
                 cbf_handle.find_column(b"depends_on")
-                depends_on.append(cbf_handle.get_value())
+                dependants[cbf_handle.get_value().decode()] = axis_name
             cbf_handle.next_row()
 
         # find the starting angles of each goniometer axis and figure out which one
         # is the scan axis (i.e. non-zero angle_increment)
         cbf_handle.find_category(b"diffrn_scan_axis")
         for i in range(cbf_handle.count_rows()):
             cbf_handle.find_column(b"axis_id")
-            axis_name = cbf_handle.get_value()
-            if axis_name.decode() not in axis_names:
+            axis_name = cbf_handle.get_value().decode()
+            if axis_name not in axis_vectors:
                 cbf_handle.next_row()
                 continue
             cbf_handle.find_column(b"angle_start")
             axis_angle = float(cbf_handle.get_value())
             cbf_handle.find_column(b"angle_increment")
             increment = float(cbf_handle.get_value())
-            angles.append(axis_angle)
-            if abs(increment) > 0:
-                assert (
-                    scan_axis is None
-                ), "More than one scan axis is defined: not currently supported"
-                scan_axis = flex.first_index(axis_names, axis_name)
+            angles[axis_name] = axis_angle
+            if increment != 0:
+                if scan_axis:
+                    raise ValueError(
```

Comment:
Should we try and fake some data to exercise this?

Reply:
Exactly my plan.